hologram_dashboard/proto++/protovalue.h

```diff
@@ -0,0 +1,111 @@
+/*
+*   Copyright 2020 Google LLC
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*       https://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*/
+
+#pragma once
+#include <iostream>
+#include <gtest/gtest.h>
+#include <assert.h>
+#include <fstream>
+#include <google/protobuf/text_format.h>
+#include <vector>
+#include <memory>
+#include "absl/types/variant.h"
+
+namespace wireless_android_play_analytics {
+
+class ProtoValue {
+
+public:
+    // The only place where name can be set publicly since once set it should
+    // not be changed.
```

Comment:
I am a bit confused by this comment. What does name refer to (field name)? Do you mean that field_name is immutable once set?